### PR TITLE
Update rampfitting docs to clarify error propagation

### LINKED ISSUE #1874: Need error propagation documentation
Currently being tracked as jira ticket: https://jira.stsci.edu/browse/JP-230:

There is not documentation about the current implementation of the error propagation.

### PR DESCRIPTION
**Pull request by dmggh**
Updates for error propagation documentation. Fixes #1874 and JIRA ticket JP-230.

### CODE REVIEW

File: docs/jwst/ramp_fitting/main.rst [hunk 1]
```diff
@@ -28,65 +28,95 @@ using the array-based functionality of numpy.  The size of the block depends
...
+segment having only a single group, the count rate will be calculated to be the
+value of the science data in that group divided by the group time.  If a ramp
+has a segment having a single group, and at least one other segment having more
+than one good group, only data from the segment(s) having more than a single 
+good group will be used to calculate the count rate.
+
+
+The data is checked for ramps in which there is good data in the first group, 
+but all first differences for the ramp are undefined because there are too few
+good groups past the first. For such ramps the first differences will be set to
```
Comment (philhodge) [inline docs/jwst/ramp_fitting/main.rst]: Do you specify anywhere what "too few good groups" means?

File: docs/jwst/ramp_fitting/main.rst [hunk 2]
```diff
@@ -28,65 +28,95 @@ using the array-based functionality of numpy.  The size of the block depends
...
-but with the only nonzero values being those whose pixel locations are flagged in
-the input GROUPDQ as cosmic ray hits. The array is compressed, removing all
-groups in which all the values are 0 for pixels having at least one group with a
-non-zero magnitude. The order of the cosmic rays within the ramp is preserved.
+VAR_POISSON, and VAR_RNOISE which contain the slopes, uncertainties in the
+slopes, y-intercept, uncertainty in the y-intercept, fitting weights, the
+variance of the slope due to poisson noise only, and the variance of the slope
+due to read noise only for each segment of each pixel. (Calculation of the two
+variance arrays requires retrieving readnoise and gain values from their
+respective reference files.)  The y-intercept refers to the result of the fit
```
Comment (philhodge) [inline docs/jwst/ramp_fitting/main.rst]: Does the user need to know that readnoise and gain were used to calculate the variances?  A reader might wonder whether he/she would need the readnoise and gain reference files in order to calculate the variances from the information in the VAR_POISSON and VAR_READNOISE values.

File: docs/jwst/ramp_fitting/main.rst [hunk 3]
```diff
@@ -217,3 +247,28 @@ The ramp fitting step has three optional arguments that can be set by the user:
...
+output product. This combined variance of the exposure-level slope is the sum
+of the variance of the slope due to the Poisson noise and the variance of the 
+slope due to the read noise. These two variances are also separately written
+to the extensions VAR_POISSON and VAR_RNOISE in the primary output.
+
+
+At the integration-level, the variance of the integration-level slope due to
+Poisson noise is written to the VAR_POISSON extension in the
+integration-specific product, and the variance of the integration-level slope
+due to read noise noise is written to the VAR_RNOISE extension.  The combined
```
Comment (philhodge) [inline docs/jwst/ramp_fitting/main.rst]: typo:  "noise noise"

**Comment by dmggh** [comment]
Based on reviewing Daryl's proposal planning document, and discussing with Phil, it seems that using TGROUP for the group time when only the first group is good.

**Pull request merged**